SingularityService/src/main/java/com/hubspot/singularity/data/SingularityValidator.java

```diff
@@ -377,6 +373,34 @@ public SingularityRequest checkSingularityRequest(
       .build();
   }
 
+  private void checkRequestAgainstMaxScale(SingularityRequest request) {
```

Comment:
I wasn't sure how to implement this logic more concisely such that users are still told specifically which max scale config they are violating, without doing anything hacky